src/cli.rs

```diff
@@ -0,0 +1,733 @@
+    ///   - Find empty directories:
+    ///       fd --type empty --type directory
+    ///       fd -te -td"
+    #[clap(
+        long = "type",
+        short = 't',
+        action,
+        value_name = "filetype",
+        hide_possible_values = true,
+        value_enum,
+        verbatim_doc_comment
+    )]
+    pub filetype: Option<Vec<FileType>>,
+    /// Filter by file extension
+    ///
+    /// (Additionally) filter search results by their file extension. Multiple
+    /// allowable file extensions can be specified.
+    /// If you want to search for files without extension,
+    /// you can use the regex '^[^.]+$' as a normal search pattern.
+    #[clap(long = "extension", short = 'e', value_name = "ext", action)]
+    pub extensions: Option<Vec<String>>,
+
+    #[clap(flatten)]
+    pub exec: Exec,
+
+    /// Max number of arguments to run as a batch with -X
+    ///
+    /// Maximum number of arguments to pass to the command given with -X.
+    /// If the number of results is greater than the given size,
+    /// the command given with -X is run again with remaining arguments.
+    /// A batch size of zero means there is no limit (default), but note
+    /// that batching might still happen due to OS restrictions on the
+    /// maximum length of command lines.
+    #[clap(
+        long,
+        value_name = "size",
+        hide_short_help = true,
+        requires("exec-batch"),
+        value_parser = value_parser!(usize),
+        default_value_t
+    )]
+    pub batch_size: usize,
+    /// Exclude entries that match the given glob pattern
+    ///
+    /// Exclude files/directories that match the given glob pattern. This overrides any other
+    /// ignore logic. Multiple exclude patterns can be specified.
+    ///
+    /// Examples:
+    ///   --exclude '*.pyc'
+    ///   --exclude node_modules
+    #[clap(long, short = 'E', value_name = "pattern", verbatim_doc_comment)]
+    pub exclude: Vec<String>,
+    /// Add custom ignore-file in '.gitignore' format
+    ///
+    /// Add a custom ignore-file in '.gitignore' format. These files have a low
+    /// precedence.
+    #[clap(long, value_name = "path", action, hide_short_help = true)]
+    pub ignore_file: Vec<PathBuf>,
+    /// When to use colors
+    ///
+    /// 'auto':      show colors if the output goes to an interactive console (default)
+    /// 'never':     do not use colorized output
+    /// 'always':    always use colorized output
+    #[clap(
+        long,
+        short = 'c',
+        value_enum,
+        default_value_t = ColorWhen::Auto,
+        value_name = "when",
+        hide_possible_values = true,
+        verbatim_doc_comment
+    )]
+    pub color: ColorWhen,
+    /// Set number of threads
+    ///
+    /// Set number of threads to use for searching & executing (default: number
+    /// of available CPU cores)
+    #[clap(long, short = 'j', value_name = "num", hide_short_help = true, value_parser = RangedU64ValueParser::<usize>::from(1..))]
```

Comment:
nit: did `value_parser = 1..` not work?

Another syntax option is `value_parser!(usize).range(1..)`

Oh, is it because those use a `i64` under the hood and you lose out out on a bit for signed?  Man, I wish I had that many cores :)

Reply:
Yeah, might be worth re-evaluating if we could not use another type here :-)

Reply:
I can't imagine you would need more than  `u32::MAX` (or even `i32::MAX`) threads.

Reply:
well, I missed that `1..` only works for i64 and u64. And since `usize` can actually be smaller  (for example on 32 bit machines) being able to use `1..` to get a u64 would add complexity elsewhere.